sdjournal/journal.go

```diff
@@ -1118,3 +1136,26 @@ func (j *Journal) GetCatalog() (string, error) {
 
 	return catalog, nil
 }
+
+// GetBootID get systemd boot id
+func (j *Journal) GetBootID() (string, error) {
+	sd_id128_get_boot, err := getFunction("sd_id128_get_boot")
+	if err != nil {
+		return "", err
+	}
+
+	var boot_id C.sd_id128_t
+	r := C.my_sd_id128_get_boot(sd_id128_get_boot, &boot_id)
+	if r < 0 {
+		return "", fmt.Errorf("failed to get boot id: %s", syscall.Errno(-r).Error())
+	}
+
+	sd_id128_to_string, err := getFunction("sd_id128_to_string")
+	c := (*C.char)(C.malloc(32))
```

Comment:
I think this constant should be `33`, i.e. `SD_ID128_STRING_MAX`, in order to account for the NUL terminator.